auto_dev/commands/publish.py

```diff
@@ -0,0 +1,86 @@
+"""This module contains the logic for the publish command."""
+
+from pathlib import Path
+
+import rich_click as click
+from aea.configurations.base import PublicId
+
+from auto_dev.base import build_cli
+from auto_dev.utils import change_dir
+from auto_dev.constants import AGENT_PUBLISHED_SUCCESS_MSG
+from auto_dev.exceptions import OperationError
+from auto_dev.services.package_manager.index import PackageManager
+
+
+cli = build_cli()
+
+
+@cli.command()
+@click.argument(
+    "public_id",
+    type=PublicId.from_str,
+    required=False,
+)
+@click.option(
+    "--force",
+    is_flag=True,
+    help="Force overwrite if package already exists",
+    default=False,
+)
+@click.pass_context
+def publish(ctx, public_id: str = None, force: bool = False) -> None:
+    """
+    Publish an agent to the local registry.
+
+    Args:
+        public_id: Optional. The public_id of the agent in the open-autonmy format i.e. `author/agent`.
+                  If not provided, assumes you're inside the agent directory.
+        force: If True, will overwrite existing package.
+
+    Example usage:
+        From parent directory: `adev publish author/agent`
+        From agent directory: `adev publish`
+        With force: `adev publish --force`
+    """
+    verbose = ctx.obj["VERBOSE"]
+    logger = ctx.obj["LOGGER"]
+
+    if public_id:
+        logger.info(f"Publishing agent {public_id}")
+    else:
+        logger.info("Publishing agent from current directory")
+
+    try:
+        package_manager = PackageManager(verbose=verbose)
+
+        # If we're given a public_id, we need to cd into that directory first
+        if public_id:
+            if isinstance(public_id, str):
+                public_id = PublicId.from_str(public_id)
+
+            agent_path = Path(public_id.name)
+            if not agent_path.exists():
```

Comment:
we have the agent manager for this now!
It is in create and has functions for this logic, that will take out like 10 lines i think